docs/content/docs/internals/data_lineage.md

```diff
@@ -0,0 +1,74 @@
+Flink native lineage interfaces are defined in two layers. The first layer is the generic interface for all Flink jobs and connector, and the second layer defines
+the extended interfaces for Table and DataStream independently. The interface and class relationship are defined in the diagram below.
+
+{{< img src="/fig/lineage_interfaces.png" alt="Lineage Data Model" width="80%">}}
+
+By default, Table related lineage interfaces or classes are mainly used in Flink Table Runtime, thus Flink users doesn't need to touch these interfaces. Flink community will gradually support all
+of common connectors, such as Kafka, JDBC, Cassandra, Hive and so on. If you have customized connector defined, you need to have customized source/sink implements the LineageVertexProvider interface.
+Within a LineageVertex, a list of Lineage Dataset is defined as metadata for Flink source/sink. 
+
+
+```java
+@PublicEvolving
+public interface LineageVertexProvider {
+  LineageVertex getLineageVertex();
+}
+```
+
+For the interface details, please refer to [FLIP-314](https://cwiki.apache.org/confluence/display/FLINK/FLIP-314%3A+Support+Customized+Job+Lineage+Listener).
+
+# Naming Conventions
+For each of Lineage Dataset, we need to define its own name and namespace to distinguish different data store and corresponding instance used in the connector of a Flink application. 
+
+| Data Store | Connector Type  | Namespace                              | Name                                                     | 
+|------------|-----------------|----------------------------------------|----------------------------------------------------------|
+| Kafka      | Kafka Connector | kafka://{bootstrap server host}:{port} | topic                                                    |
+| MySQL      | JDBC Connector  | mysql://{host}:{port}                  | {database}.{table}                                       | 
+| Sql Server | JDBC Connector  | sqlserver://{host}:{port}              | {database}.{table}                                       | 
+| Postgres   | JDBC Connector  | postgres://{host}:{port}               | {database}.{schema}.{table}                              | 
+| Oracle     | JDBC Connector  | oracle://{host}:{port}                 | {serviceName}.{schema}.{table} or {sid}.{schema}.{table} | 
+| Trino      | JDBC Connector  | trino://{host}:{port}                  | {catalog}.{schema}.{table}                               | 
+| OceanBase  | JDBC Connector  | oceanbase://{host}:{port}              | {database}.{table}                                       | 
+| DB2        | JDBC Connector  | db2://{host}:{port}                    | {database}.{table}                                       | 
+| CrateDB    | JDBC Connector  | cratedb://{host}:{port}                | {database}.{table}                                       | 
+
+It is a running table. More and more naming info will be added after lineage integration is finished for a specific connector.
```

Comment:
I am struggling with the sentence "It is a running table. More and more naming info will be added after lineage integration is finished for a specific connector." I am not sure hat you trying to say. when you say running table , do you mean that lineage relates to how data flows though a dynamic table at runtime.
I am not sure what "more an more naming" means. I assume when the connector adds the lineage capability, it associates a name with the Lineage Dataset. Is there more we need to say around this?

I wonder if the connector information should be authored in the appropriate connector repo and brought into the core Flink docs.